flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/runtime/SavepointEnvironment.java

```diff
@@ -198,6 +199,11 @@ public GlobalAggregateManager getGlobalAggregateManager() {
 		throw new UnsupportedOperationException(ERROR_MSG);
 	}
 
+	@Override
+	public Map<String, ExternalResourceDriver> getExternalResourceDrivers() {
+		return Collections.emptyMap();
+	}
```

Comment:
Should we throw an `UnsupportedOperationException` here?

Reply:
I did throw an exception but I found error in `SavepointWriterITCase `. It seems currently the 
 `SavepointEnvironment` would still be used to construct a `StreamingRuntimeContext`. I'm not familiar with savepoint mechanism. But I guess it is an issue because the constructor of `SavepointEnvironment` already took a `RuntimeContext` instance. It does not make sense to contract `RuntimeContext` from it. Thus, I currently choose to return `emptyMap` here.

Reply:
After an offline talk with @sjwiesman , I believe it probably makes sense to return emptyMap here.
The root reason of the failure of the `ITCase` is that in `BoundedOneInputStreamTaskRunner.mapPartition` we start a new `BoundedStreamTask`  with `SavepointEnvironment`. We have to do this is because there is no way of accessing the actual execution environment from within the dataset API. Eventually, when there is a bounded datastream api, all of this will go away. So, at the moment, I think it would be ok to return `Collections.emptyMap()`. @sjwiesman Please correct me if I don't express it correctly.

Reply:
+1 for the empty map

Reply:
Make sense to me.
I would also suggest to comment some brief explanation here.